esp-hal/src/lib.rs

```diff
@@ -525,3 +525,21 @@ fn hal_main(a0: usize, a1: usize, a2: usize) -> ! {
 unsafe extern "C" fn stack_chk_fail() {
     panic!("Stack corruption detected");
 }
+
+/// Helper macro for checking doctest code snippets
+#[cfg(not(host_os = "windows"))]
+#[macro_export]
+macro_rules! before_snippet {
+    () => {
+        core::include_str!(concat!(env!("CARGO_MANIFEST_DIR"), "/doc-helper/before"))
+    };
+}
+
+/// Helper macro for checking doctest code snippets
+#[cfg(host_os = "windows")]
```

Comment:
for this to work we'd need the same as https://github.com/dtolnay/rustversion/blob/c1314d0d220720c5eae65fe8ae376f439d69aee6/build/build.rs#L86-L89 in our build.rs

But I think the macro is a really nice solution